pkg/gitreceive/build.go

```diff
@@ -138,8 +138,21 @@ func build(
 
 	var pod *api.Pod
 	var buildPodName string
+	image := appName
 	if usingDockerfile {
 		buildPodName = dockerBuilderPodName(appName, gitSha.Short())
+		registryLocation := conf.RegistryLocation
+		registryEnv := make(map[string]string)
+		if registryLocation != "on-cluster" {
+			registryEnv, err = getRegistryDetails(kubeClient, &image, registryLocation, conf.PodNamespace, conf.RegistrySecretPrefix)
+			if err != nil {
+				return fmt.Errorf("error getting private registry details %s", err)
+			}
+			image = image + ":git-" + gitSha.Short()
+		}
+		registryEnv["DEIS_REGISTRY_PROXY_PORT"] = conf.RegistryProxyPort
```

Comment:
DEIS_REGISTRY_PROXY_PORT was the initial env which i used to pass the port of the proy service and we  reverted the dockerbuilder and controller pr but din't the builder as we thought we want to bring the feature anyway.
Now, i see you are updating the DEIS_REGISTRY_SERVICE_PORT/HOST itself.
So either you or me has to rebase depending on who merges first 😄